src/calculateAverage.js

```diff
@@ -32,12 +32,58 @@
  *
  */
 function applyCalculateAverage() {
+  function calculateAverageOfArray(array) {
+    const initialObject = {
+      sum: 0,
+      counter: 0,
+      average: 0
+    };
+    const arrayInformation = array.reduce((accumulator, currentValue) => {
+      return {
+        sum: accumulator.sum + currentValue,
+        counter: accumulator.counter + 1,
+        average: (accumulator.sum + currentValue) / (accumulator.counter + 1)
+      };
+    }, initialObject);
```

Comment:
this 'reduce' is way too complex. 

All you need is calculate sum and then return sum/array.length.

Reply:
Try rewriting and compare if code is easier to read.